### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by nicolejkeeney**
This PR sets up a framework for improving and fleshing out documentation of climakitae. It's still a bit bare-bones now, and I have yet to add the entire section on scientific toolkits (i.e how to use the various explore panels, timeseries stuff, etc). Let me know what you all think of the documentation at this stage

In order to see how the readthedocs page will look (current version [here](https://climakitae.readthedocs.io/en/latest/)), you will need to **build the documentation locally**. You are welcome to do this, and there's instructions for how to do so [here](https://github.com/cal-adapt/climakitae/blob/documentation/docs/README.md). This is the only way you'll be able to preview how the readthedocs site will look before we merge this branch into main. 

If you'd like to get a sense for how this page will look but are uninterested in building the docs locally, I've provided some screenshots at the bottom. 

**How to review**: 
1) Observe (within GitHub is fine) the markdown in the "Working with our Data" rst file: https://github.com/cal-adapt/climakitae/blob/documentation/docs/data.rst
2) Provide some feedback on the design, either by [building the docs yourself locally](https://github.com/cal-adapt/climakitae/blob/documentation/docs/README.md) or just observing the screenshots included in below. 

**Homepage:**
Homepage

**About our data:**
data

**climakitae API:**
API page

**example of function documentation:**
documentation example

### CODE REVIEW

**Review by elehmer — APPROVED** [review]
This all looks good.

**Review by vicford — CHANGES_REQUESTED** [review]
This looks **amazing**, @nicolejkeeney, really excited to see how this builds upon in the future!

A couple of things, mainly questions: 
1. building locally didn't work for me, so just reviewing based on the docs provided in the PR
2. In the first instance where app = ck.Application there are no parentheses on the Application object (in the app.select description), but clearly shows in the core functions page, and what we do on the Hub too. If it doesn't make a difference (which upon testing on the hub, it does), I do think we should still be consistent about how we display that call in our docs. 
3. In "Directly modifying the location and selections attributes" section where things are highlighted, is it "selectors" or "selections"?
4. In "Using a csv config file", we appropriately state that the headers must be exactly as they are in the following example, meaning that they must be labeled precisely as such, and/or the order must follow that format? Can be more explicit here on this

**Comment by nicolejkeeney** [comment]
> 2. In the first instance where app = ck.Application there are no parentheses on the Application object (in the app.select description), but clearly shows in the core functions page, and what we do on the Hub too. If it doesn't make a difference (which upon testing on the hub, it does), I do think we should still be consistent about how we display that call in our docs.

Good point, I added back in some parentheses and I think I fixed the issue! Thanks for spotting that.

**Comment by nicolejkeeney** [comment]
> 3. In "Directly modifying the location and selections attributes" section where things are highlighted, is it "selectors" or "selections"?

It is "selections", which is a DataSelector object-- a bit confusing I guess! But selections is correct.

**Comment by nicolejkeeney** [comment]
> 4. In "Using a csv config file", we appropriately state that the headers must be exactly as they are in the following example, meaning that they must be labeled precisely as such, and/or the order must follow that format? Can be more explicit here on this

I agree that that section is confusing. I modified the wording a bit to see if that cleared it up?

**Review by vicford — APPROVED** [review]
Looks great!

File: docs/data.rst [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+**********************
+Working with our Data
+**********************
+Data from the Cal-Adapt Analytics Engine can be retrieved, subsetted, visualized, and 
+exported using the *climakitae library*. 
+
+
+Retrieve and subset the data
+#############################
+In this section we will detail the various methods to retrieve and subset the catalog data.
```
Comment (gdicecco) [inline docs/data.rst:12]: Here I suggest we either link to the data catalog page on the website (https://analytics.cal-adapt.org/data/) or link & condense some of that info here to make this able to stand on its own (i.e. mention somewhere what kind of data is in the catalog)

File: docs/data.rst [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+Retrieve and subset the data
+#############################
+In this section we will detail the various methods to retrieve and subset the catalog data. 
+
+Use the app.select() panel GUI 
+*********************************
+If you are working in a Jupyter notebook environment, you can view and set your data and location 
+options in the :py:func:`climakitae.Application.select()` GUI (graphical user interface). This GUI also provides a visual overview of the various 
+datasets available in the AE data catalog. Using this GUI, you can chose what dataset you'd like to 
+retrieve-- chosing a variable, timeslice, resolution, etc.-- and the location for which you'd like to
```
Comment (gdicecco) [inline docs/data.rst:19]: Typo - 'choosing a variable...'

File: docs/data.rst [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+     - Yes
+     - hourly
+     - 9 km
+     - (2005, 2025)
+     - states 
+     - CA
+
+Read the data into memory 
+###########################
+The data is retrieved as lazily loaded Dask arrays until you chose to read the data into
```
Comment (gdicecco) [inline docs/data.rst:112]: typo *choose to read....

File: docs/data.rst [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+   app.view(data)
+
+You can also set the colormap and size of the output visualization using the function arguments; see 
+the documentation in the API for more information. 
+
+Export the data 
+################
+To export the data, first chose the filetype you want to export the data to using the 
+:py:func:`climakitae.Application.export_as()` dropdown menu. This will allow you to choose 
+between three options: netcdf, csv, and geotiff.::
```
Comment (gdicecco) [inline docs/data.rst:136]: formatting thing - I don't think we need the . after geotiff here?

File: docs/data.rst [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+   app.export_as() 
+
+After selecting your desired output filetype, input the data you want to export and the
```
Comment (gdicecco) [inline docs/data.rst:140]: To be thorough here, it would be nice to repeat some of the markdown Beth wrote describing what data structures are suited to each export type (at a minimum that netcdf is most flexible, but csv is very restricted).

**Review by gdicecco — APPROVED** [review]
This is an awesome start to the documentation! A few specific comments are below.

More broadly, I'm usure how I feel about the link to the Getting Started nb render on the website being featured prominently here - I think it might be hard to parse that this is from a notebook that should be run elsewhere. But, maybe this is a good thing to have for now and later instead we would want to link to e.g. a binder instance of the getting started notebook or something like that.

File: docs/data.rst (hunk 1 above)
Comment (nicolejkeeney) [reply docs/data.rst:12]: great idea!

File: docs/data.rst (hunk 5 above)
Comment (nicolejkeeney) [reply docs/data.rst:140]: good idea!
